api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreProperties.java

```diff
@@ -65,4 +67,102 @@ public BiometricEngine getBiometricEngine() {
         }
         return engine;
     }
+
+    public PixErrorHandlingService getPixErrorHandlingService() {
+        String propertyName = RegistrationCoreConstants.GP_MPI_PIX_ERROR_HANDLER_IMPLEMENTATION;
+        PixErrorHandlingService handler = null;
+        if (isPropertySet(propertyName)) {
+            handler = getComponentByPropertyName(propertyName, PixErrorHandlingService.class);
+        }
+        return handler;
+    }
+    
+    public PdqErrorHandlingService getPdqErrorHandlingService() {
+        String propertyName = RegistrationCoreConstants.GP_MPI_PDQ_ERROR_HANDLER_IMPLEMENTATION;
+        PdqErrorHandlingService handler = null;
+        if (isPropertySet(propertyName)) {
+            handler = getComponentByPropertyName(propertyName, PdqErrorHandlingService.class);
+        }
+        return handler;
+    }
+    
+    /**
+     * Get the PDQ endpoint
+     * @return
+     */
+    public String getPdqEndpoint() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_PDQ_ENDPOINT, "localhost");
+    }
+
+    /**
+     * Get the PDQ port
+     * @return
+     */
+    public Integer getPdqPort() {
+        return Integer.valueOf(Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_PDQ_PORT, "8989"));
+    }
+
+    public String getPixEndpoint() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_PIX_ENDPOINT, "localhost");
+    }
+
+    public Integer getPixPort() {
+        return Integer.valueOf(Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_PIX_PORT, "3600"));
+    }
+
+    public String getMpiUsername() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_ACCESS_USERNAME, "admin");
+    }
+
+    public String getMpiPassword() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_ACCESS_PASSWORD, "admin");
+    }
+
+    public String getShrPatientRoot() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_PATIENT_ROOT, "1.2.3.4.5.9");
+    }
+
+    public String getReceivingApplication() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_RECEIVING_APPLICATION, "CR");
+    }
+
+    public String getReceivingFacility() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_RECEIVING_FACILITY, "Clinic");
```

Comment:
I don't think the default value here make sense

Reply:
changed to "Demo Receiving Facility" and made to changed config.xml to match

Reply:
I don't think you need a default value anyways

Reply:
But theres no harm in having it right?

Reply:
The fact that setting has no effect too, you might as well not set it

Reply:
I dont understand, so in this function if there is no global property it will set the default value to this right? so there would be an effect? these defaults would show up in the HL7 messages, so it can be helpful for developers / implementers trying to figure what is going on and which field is set where, as opposed to everything being blank... doesn't it help with clarity?

Reply:
I believe blank or not, you get the same result from the MPI, so why bother to set it?

Reply:
This is true, we just tested and verified that you can send the message with these fields blank. Why we would bother to set it is to make it easy to audit. In a system where we have many openmrs instances it would be good to track where messages are coming from and where they are going to. It would make reading the transaction logs in openHIM clearer.

Reply:
But in the event you have multiple instances, then still these default values would be useless since one would still have to change them to avoid duplicates in order to serve their purpose.

Reply:
You will have to change them but if you go straight to openHIM while running the demo, you can take a look at the message and see which field goes where and it can help you to understand how the message is structured better.
For example 
`
MSH|^~\&|OpenMRS|Demo Sending Facility|Demo MPI|Demo Receiving Facility|20180803155106||ADT^A08^ADT_A01|f73e094c-9519-44f8-b5d6-fda50adce1aa|P|2.3.1|||AL
PID|||A100165^^^2.25.71280592878078638113873461180761116318&2.25.71280592878078638113873461180761116318&PI~3e4f11e0-974f-11e8-b8b1-000d3a1983eb^^^2.16.840.1.113883.4.357&2.16.840.1.113883.4.357&hl7||b3^b3^^^^^L|null, null^^^^^^U|20180207|F|||home^^^^^^L
`
vs seeing this:

`Request body
MSH|^~\&|||||20180803155106||ADT^A08^ADT_A01|f73e094c-9519-44f8-b5d6-fda50adce1aa|P|2.3.1|||AL
PID|||A100165^^^2.25.71280592878078638113873461180761116318&2.25.71280592878078638113873461180761116318&PI~3e4f11e0-974f-11e8-b8b1-000d3a1983eb^^^2.16.840.1.113883.4.357&2.16.840.1.113883.4.357&hl7||b3^b3^^^^^L|null, null^^^^^^U|20180207|F|||home^^^^^^L
`